gluonnlp/data/stream.py

```diff
@@ -162,8 +162,20 @@ def __init__(self, stream, fn):
         self._fn = fn
 
     def __iter__(self):
-        for item in iter(self._stream):
+        stream_iter = iter(self._stream)
+        try:
+            item = next(stream_iter)
+        except StopIteration:
+            return
```

Comment:
should we allow the StopIteration to throw? the code invoking `__iter__` is likely expecting an (exhausted) iterator instead of None.

Reply:
The correct behavior for a function including `yield` is to return None to notify about the end. If we raise StopIteration, pylint will fail as we must not raise StopIteration in such kind of function ;)

Reply:
R:166,15: Do not raise StopIteration in generator, use return statement instead (stop-iteration-return)

Reply:
I see. Thanks